core/tracestack.lua

```diff
@@ -0,0 +1,214 @@
+-- Represents a stack of frame objects, which describe the call-stack stack of
+-- processing of the currently processed document.
+local traceStack = {
+
+  -- Internal: Push-pop balance checking ID
+  _lastPushId = 0,
+
+  -- Stores the frame which was last popped. Reset after a push.
+  -- Helps to further specify current location in the processed document.
+  afterFrame = nil,
+
+  -- Internal: Function assigned to stack frames to convert them to human readable location
+  _frameToLocationString = function(self, skipFile, withAttrs)
+    local str
+    if skipFile or not self.file then
+      str = ""
+    else
+      str = self.file .. " "
+    end
+    if self.line then
+      str = str .. "l." .. self.line .. " "
```

Comment:
Would there by any objection if I edited this to roughly match the output format of `debug.stacktrace()` which uses `filename:N` instead of `filename l.N`, and so forth?

Reply:
I wanted to do that myself, but I wanted to conform to the existing style of previous location information.

However note, that the file is not always printed - it may not be known or (more often) it is suppressed (for example when two frame infos are on the same line and they both refer to the same file). So rewriting this naively to `str = str .. ":" .. self.line .. " "` **will** lead to rather cryptic outputs.

Reply:
Hm. Also it leaves the column info with no good home. I might not touch that, I was just wondering...

Reply:
The column is then usually formatted similarly: "`file.sile:12:34`"